### PR TITLE
Fix for color progress in challenge userprogress modal fixes issue #10907

### LINKED ISSUE #10907: challenge participant progress page should not use grey for tasks
_EDIT: There's a partial PR at https://github.com/HabitRPG/habitica/pull/10973 which a contributor could build on if they didn't want to work on this issue from scratch._

When you are looking at a challenge's page (e.g., a page such as `https://habitica.com/challenges/6c565677-65fc-400f-b39e-52bd01e4cd7a`) and you use the "View Progress Of" > "Select a Participant" menu to view the progress of a participant, completed Dailies and To-Dos should not be grey - they should show their red/orange/yellow/green/blue colour so that you can judge the participant's history of performing that task.

For example, in the screenshot below, the Daily should not be grey.
image

Whoever works on this issue should test that their change does not affect any tasks (including challenge tasks) in a player's own tasks list, or in the "Group Tasks List" screen of a guild or party with a Group Plan. For information about adding a Group Plan to a guild/party, see "Creating Test Data" in [Using Your Local Install to Modify Habitica's Website and API](https://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API)

### PR DESCRIPTION
**Pull request by gautamz07**
Fix for color progress in challenge userprogress modal fixes issue #10907

### Fixes

Fixes #10907

### Changes
basically passing a prop from `challengeMemberProgressModal.vue ` , the prop being `:showtaskcolor='true' `

@Alys have not added any tests, as i've never written any , will this be required ? please let me know :)

### CODE REVIEW

**Comment by paglias** [comment]
Hi @gautamz07  and thanks for the PR! Sorry for the delay here but we'll review the pr soon

File: website/client/components/tasks/column.vue [hunk 1]
```diff
@@ -306,6 +307,9 @@ export default {
       type: Boolean,
       default: false,
     },
+    showtaskcolor : {
+      type: Boolean
```
Comment (paglias) [inline website/client/components/tasks/column.vue]: can you give this a default value and rename it to showTaskColor? thanks!

**Review by paglias — COMMENTED** [review]
I've left one comment, but otherwise it looks good. Regarding the tests it's fine if the client doesn't have them

**Comment by paglias** [comment]
Ah, I forgot but it looks like there are some linting issues https://travis-ci.org/HabitRPG/habitica/jobs/552780174

Also, could you revert the changes to package-lock.json? (`git checkout master -- package-lock.json`)

thanks!

**Comment by gautamz07** [comment]
@paglias Thanks alot for the inputs , changes requested have been made , check my comments for details.

Just curious , how did the package-lock.json file change even through i did't change anything in it ?

Also, the coverall's is failing, any idea why ?

**Comment by paglias** [comment]
@gautamz07 thanks for the updates! I'll give it a final test later this week but it looks good

> Just curious , how did the package-lock.json file change even through i did't change anything in it ?

Eh it happens sometimes when you run `npm install` due to different operative systems / npm versions but it should only happen right after you install habitica for the first time.

> Also, the coverall's is failing, any idea why ?

to be honest coverall's being a bit buggy lately so don't worry about it, the important is that travis is green :)

**Comment by gautamz07** [comment]
> to be honest coverall's being a bit buggy lately so don't worry about it, the important is that travis is green :)

Thanks that will be great :) :+1:

**Comment by gautamz07** [comment]
@paglias any update on this ?

**Comment by paglias** [comment]
Sorry for the delay! I just tested this and it's removed the gray state from user tasks as well 
Schermata 2019-08-18 alle 17 08 50

I think the problem is because the default value is `true` and it's never set to false. I think it might be better to change the name to `alwaysShowTaskColor`, set the default value to false and then set it to true only for the challenge progress. Does it makes sense?

**Comment by paglias** [comment]
Hi @gautamz07 , do you have any update on this PR? thanks!

**Comment by paglias** [comment]
Hi @gautamz07 , I'm going to close this one but if you want to work again on this just leave a comment

**Pull request closed without merging**